sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/storage/PreferenceTokenStorage.java

```diff
@@ -0,0 +1,83 @@
+package com.microsoft.appcenter.identity.storage;
+
+import android.content.Context;
+import android.support.annotation.VisibleForTesting;
+
+import com.microsoft.appcenter.utils.context.AuthTokenContext;
+import com.microsoft.appcenter.utils.crypto.CryptoUtils;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+
+/**
+ * Storage for tokens that uses {@link SharedPreferencesManager}. Handles saving and encryption.
+ */
+public class PreferenceTokenStorage implements AuthTokenStorage {
+
+    /**
+     * {@link Context} instance.
+     */
+    private final Context mContext;
+
+    /**
+     * Default constructor.
+     *
+     * @param context {@link Context} instance.
+     */
+    PreferenceTokenStorage(Context context) {
+        mContext = context;
+    }
+
+    /**
+     * Used for authentication requests, string field for auth token.
+     */
+    @VisibleForTesting
+    static final String PREFERENCE_KEY_AUTH_TOKEN = "AppCenter.auth_token";
+
+    /**
+     * Used for distinguishing users, string field for home account id.
+     */
+    @VisibleForTesting
+    static final String PREFERENCE_KEY_HOME_ACCOUNT_ID = "AppCenter.account_id";
```

Comment:
Was it in the spec to store multiple tokens for different users? I was thinking we would just store one token at a time

Reply:
I didn't implement multiple tokens. Just storing the last account id to distinguish new users

Reply:
ah it's account id. got it.